go/core/action.go

```diff
@@ -178,7 +215,10 @@ func (a *Action[In, Out, Stream]) Run(ctx context.Context, input In, cb func(con
 			}
 			var output Out
 			if err == nil {
-				output, err = a.fn(ctx, input, cb)
+				dispatch := chainMiddleware(a.middleware...)
```

Comment:
Maybe I'm missing something, but I don't think we need to pass down the middlewares as a separate thing.

Say we changed middleware handlers so they have the same type as the action function itself, by including the callback argument.

Then chaining the handlers followed by the action function doesn't have to happen here, it could happen higher up.
So instead of
```
DefineAction(..., Middlewares(h1, h2), f)
```
we could write
```
DefineAction(..., h1(h2(f)))
```
This effectively removes the machinery of middlewares from the implementation. A middleware handler is just a function from action-functions to action-functions. We can keep the `MiddlewareHandler` type and perhaps provide a couple of common ones, but I don't think we need the rest of it.